api/apps/api/src/modules/scenarios/input-files/puvspr.dat.service.ts

```diff
@@ -1,27 +1,48 @@
-import { Repository } from 'typeorm';
+import { Connection, getConnection } from 'typeorm';
 import { Injectable } from '@nestjs/common';
-import { InjectRepository } from '@nestjs/typeorm';
-
-import { ScenarioPuvsprGeoEntity } from '@marxan/scenario-puvspr';
 import { DbConnections } from '@marxan-api/ormconfig.connections';
-
+import { InjectConnection } from '@nestjs/typeorm';
 @Injectable()
 export class PuvsprDatService {
   constructor(
-    @InjectRepository(ScenarioPuvsprGeoEntity, DbConnections.geoprocessingDB)
-    private readonly puvsprRepo: Repository<ScenarioPuvsprGeoEntity>,
+    @InjectConnection(DbConnections.geoprocessingDB)
+    private readonly connection: Connection,
```

Comment:
you can just use `puvsprRepo.query` instead creating new connection.

Reply:
yep, I know - I did so because with this PR the `puvsprRepo` should become redundant and we should be able to drop it altogether: I left it in the current PR in order not to change too much in case the new approach ends up not working out in practice, but all going well we should be able to drop the repo.